extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -636,27 +634,38 @@ export class AmpA4A extends AMP.BaseElement {
     // signature, then the creative is valid AMP.
     /** @type {!AllServicesCryptoKeysDef} */
     const keyInfoSetPromises = this.win.ampA4aValidationKeys;
+    // Track if verification found, as it will ensure that promises yet to
+    // resolve will "cancel" as soon as possible saving unnecessary resource
+    // allocation.
+    let verified = false;
     return some(keyInfoSetPromises.map(keyInfoSetPromise => {
       // Resolve Promise into an object containing a 'keys' field, which
       // is an Array of Promises of signing keys.  *whew*
       return keyInfoSetPromise.then(keyInfoSet => {
         // As long as any one individual key of a particular signing
         // service, keyInfoPromise, can verify the signature, then the
         // creative is valid AMP.
+        if (verified) {
+          return Promise.reject('noop');
+        }
         return some(keyInfoSet.keys.map(keyInfoPromise => {
           // Resolve Promise into signing key.
           return keyInfoPromise.then(keyInfo => {
+            if (verified) {
+              return Promise.reject('noop');
+            }
             if (!keyInfo) {
               return Promise.reject('Promise resolved to null key.');
             }
             const signatureVerifyStartTime = this.getNow_();
             // If the key exists, try verifying with it.
-            return verifySignature(
+            return this.crypto_.verifySignature(
                 new Uint8Array(creative),
                 signature,
                 keyInfo)
                 .then(isValid => {
```

Comment:
the indentation is a bit off, is there enough space to fit all params in one line?